services/s3/src/main/java/software/amazon/awssdk/services/s3/checksums/ChecksumValidatingInputStream.java

```diff
@@ -162,26 +162,15 @@ public void close() throws IOException {
         inputStream.close();
     }
 
-    /**
-     * Gets the stream's checksum as an integer.
-     *
-     * @return checksum.
-     */
-    public int getStreamChecksum() {
-        ByteBuffer bb = ByteBuffer.wrap(streamChecksum);
-        return bb.getInt();
-    }
-
     private void validateAndThrow() {
-        int streamChecksumInt = getStreamChecksum();
         if (computedChecksum == null) {
-            computedChecksum = ByteBuffer.wrap(checkSum.getChecksumBytes()).getInt();
+            computedChecksum = checkSum.getChecksumBytes();
         }
 
-        if (streamChecksumInt != computedChecksum) {
+        if (!Arrays.equals(computedChecksum, streamChecksum)) {
             throw SdkClientException.builder().message(
-                String.format("Data read has a different checksum than expected. Was %d, but expected %d",
-                              computedChecksum, streamChecksumInt)).build();
+                String.format("Data read has a different checksum than expected. Was %s, but expected %s",
+                              Arrays.toString(computedChecksum), Arrays.toString(streamChecksum))).build();
```

Comment:
minor: should we represent as a hex string instead? Would make the error message shorter and bit easier to digest easier to digest

Reply:
+1